src/main/java/com/openmrs/migrator/Migrator.java

```diff
@@ -0,0 +1,193 @@
+package com.openmrs.migrator;
+
+import com.openmrs.migrator.core.services.BootstrapService;
+import com.openmrs.migrator.core.services.DataBaseService;
+import com.openmrs.migrator.core.services.PDIService;
+import com.openmrs.migrator.core.utilities.ConsoleUtils;
+import com.openmrs.migrator.core.utilities.FileIOUtilities;
+import java.io.Console;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import org.pentaho.di.core.exception.KettleException;
+import org.springframework.beans.factory.annotation.Autowired;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(name = "migrator", mixinStandardHelpOptions = true)
+public class Migrator implements Callable<Optional<Void>> {
+
+  private PDIService pdiService;
+
+  private FileIOUtilities fileIOUtilities;
+
+  private BootstrapService bootstrapService;
+
+  private DataBaseService dataBaseService;
+
+  private Console console;
+
+  private String[] jobs = {"pdiresources/jobs/merge-patient-job.kjb"};
+
+  private List<String> dirList =
+      Arrays.asList(
+          "input/",
+          "output/",
+          "config/",
+          "pdiresources/",
+          "pdiresources/transformations/",
+          "pdiresources/jobs/");
+
+  @Option(
+      names = {"run"},
+      description = "runs the migrator job(s)")
+  private boolean run;
+
+  @Option(
+      names = {"setup"},
+      description = "setups the migrator tool")
+  private boolean setup;
+
+  @Autowired
+  public Migrator(
+      Console console,
+      PDIService pdiService,
+      FileIOUtilities fileIOUtilities,
+      BootstrapService bootstrapService,
+      DataBaseService dataBaseService) {
+    this.pdiService = pdiService;
+    this.fileIOUtilities = fileIOUtilities;
+    this.bootstrapService = bootstrapService;
+    this.dataBaseService = dataBaseService;
+    this.console = console;
+  }
+
+  @Override
+  public Optional<Void> call() throws IOException, InterruptedException {
+
+    if (setup) {
+      executeSetupCommand();
+      fileIOUtilities.fillConfigFile();
```

Comment:
The reading and processing of the config/settings file falls under ticket 100. We will have a service handle that.

I think lets rather remove this and use the service from MEPTS-100 rather. It is better we separate generic fileIO utilities (such as read/write files, etc...) from config specific logic

Reply:
done